gimme_aws_creds/okta.py

```diff
@@ -423,6 +426,98 @@ def _login_input_webauthn_challenge(self, state_token, factor):
         if 'sessionToken' in response_data:
             return {'stateToken': None, 'sessionToken': response_data['sessionToken'], 'apiResponse': response_data}
 
+    def _login_duo_challenge(self, state_token, factor):
+        """ Duo MFA challenge """
+
+        if factor['factorType'] is None:
+            # Prompt user for which Duo factor to use
+            raise duo.FactorRequired(id, state_token)
+
+        if factor['factorType'] == "passcode" and not passcode:
+            raise duo.PasscodeRequired(fid, state_token)
+
+        path = '/authn/factors/{fid}/verify'.format(fid=factor['id'])
+        data = {'fid': factor['id'],
+                'stateToken': state_token}
+
+        response = self._http_client.post(factor['_links']['verify']['href'],
+            params={'rememberDevice': self._remember_device},
+            json={'stateToken': state_token},
+            headers=self._get_headers(),
+            verify=self._verify_ssl_certs
+        )
+        response_data = response.json()
+        verification = response_data['_embedded']['factor']['_embedded']['verification']
+
+        auth = None
+        duo_client = duo.Duo(verification, state_token, factor['factorType'])
+        if factor['factorType'] == "web":
+            # Duo Web via local browser
+            self.ui.info("Duo required; opening browser...")
+            proc = Process(target=duo_client.trigger_web_duo)
+            proc.start()
+            time.sleep(2)
+            webbrowser.open_new('http://127.0.0.1:65432/duo.html')
+        elif factor['factorType'] == "passcode":
+            # Duo auth with OTP code without a browser
+            self.ui.info("Duo required; using OTP...")
+            auth = duo_client.trigger_duo(passcode=passcode)
+        else:
+            # Duo Auth without the browser
+            self.ui.info("Duo required; check your phone... 📱")
```

Comment:
Emoji's aren't supported in all consoles, and create weird character artifacts in the ones that don't.  Lets remove this guy.

Reply:
Removed emoji